### PR TITLE
feat(stream-list): Creates a stream-list component

### PR DESCRIPTION
**Pull request by kennyw12**
Some open questions we had around the API design:

1. Should ListItems appear in reversed order? I.e. does 
```jsx
<StreamList>
  <ListItem>Item a</ListItem>
  <ListItem>Item b</ListItem>
```
render as:
```
Item b
Item a
```
This is so that new data appears on the top of the list.

2. Should the stream list be component be a `StreamList` (i.e. a separate component than `GroupList`)? Or should the stream list be a `GroupList` with `stream={true}` (or a similar property)?

3. The text on the button: we have two props - `singularNewItemText` and `pluralNewItemsText`. So the button would say "1 {singularNewItemText}" or "4 {pluralNewItemsText}" (e.g. "1 new item", "4 new items"). If not supplied, we provide the default values of "new item"/"new items". Does this seem like a necessary/reasonable way of making the text modular? Do you have a better idea for prop names?

### CODE REVIEW

File: src/pivotal-ui-react/stream-list/stream-list.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    );
+  }
+});
+
+export const StreamList = React.createClass({
+  mixins: [AnimationMixin],
+
+  propTypes: {
+    singularNewItemText: React.PropTypes.string.isRequired,
+    pluralNewItemsText: React.PropTypes.string.isRequired
```
Comment (matt-royal) [inline src/pivotal-ui-react/stream-list/stream-list.js]: Since these values are defaulted, these props should not be required.

**Comment by matt-royal** [comment]
I just pushed a change to remove the "required" singularNewItemText and pluralNewItemsText. However, we're still missing documentation in a .scss file, so this will not appear in the styleguide. This also raises the question about whether we need to support beta components for this to make it into the guide.

**Comment by matt-royal** [comment]
I talked to @stubbornella. We're not going to treat this as a Beta component, so just fix the lack of styleguide documentation, please.

**Comment by matt-royal** [comment]
Don't know how I missed the styleguide documentation. This is ready to accept.

**Comment by stubbornella** [comment]
Yes, new data should appear at the top of the list.

File: src/pivotal-ui-react/stream-list/stream-list.js (hunk 1 above)
Comment (stubbornella) [inline src/pivotal-ui-react/stream-list/stream-list.js]: @matt-royal should we block merging on this PR until the .isRequired lines are removed?

**Pull request merged**